dll/win32/shell32/wine/control.c

```diff
@@ -736,6 +737,96 @@ static	void	Control_DoWindow(CPanel* panel, HWND hWnd, HINSTANCE hInst)
 }
 #endif
 
+#ifdef __REACTOS__
+
+/** Structure for in and out data when
+ *  search for the cpl dialog of first instance
+ */
+typedef struct tagAppDlgFindData
+{
+    PCWSTR  szAppFile;  /**< Full path to applet library as second search value */
+    HWND    hDlgResult; /**< Returned dialog handle or NULL if not found */
+} AppDlgFindData;
+
+/**
+ * This function searches for an applet library and
+ * compares the file paths.
+ * @param hwnd The handle of the applet dialog
+ * @param szAppFile Full path of applet library
+ * @return TRUE if correct applet library otherwise FALSE
+ */
+static BOOL
+Control_FindAndCheckAppFile(
+    _In_ HWND   hwnd,
+    _In_ PCWSTR szAppFile
+)
+{
+    DWORD dwProcessId;
+    GetWindowThreadProcessId(hwnd, &dwProcessId);
+    if (dwProcessId)
+    {
+        HANDLE hProcess;
+        hProcess = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, dwProcessId);
+        if (hProcess)
+        {
+            HMODULE hMods[1024];
+            WCHAR szModName[MAX_PATH];
+            DWORD cbNeeded;
+
+            if (EnumProcessModules(hProcess, hMods, sizeof(hMods), &cbNeeded))
+            {
+
+                for (unsigned int i = 0; i < (cbNeeded / sizeof(HMODULE)); i++)
+                {
+                    // Get the full path to the module's file.
+                    if (GetModuleFileNameExW(hProcess, hMods[i], szModName, _countof(szModName)))
+                    {
+                        if (_wcsicmp(szModName, szAppFile) == 0)
+                        {
+                            return TRUE;
+                        }
+                    }
+                }
+            }
+            CloseHandle(hProcess);
+        }
+    }
+    return FALSE;
+}
+
+
+/**
+ * Callback function to search applet dialog
+ * @param hwnd A handle to a top-level window.
+ * @param lParam Pointer of AppDlgFindData
+ * @return TRUE: continue enumeration, FALSE: stop enumeration
+ */
+static BOOL CALLBACK
+Control_EnumWinProc(
```

Comment:
It seems you have branched from other PR branch, so this PR contains unrelated changes.

Reply:
How to fix:
- Do a hard reset to the master branch state
- Cherry-pick new commits on top of it, and do a force push